internal/service/mongodbemployeeaccessgrant/resource.go

```diff
@@ -99,15 +107,21 @@ func (r *rs) createOrUpdate(ctx context.Context, tfModelFunc func(context.Contex
 	}
 	atlasReq, err := NewAtlasReq(&tfModel)
 	if err != nil {
-		diagnostics.AddError(errorCreate, err.Error())
+		diagnostics.AddError(errorCreateUpdate, err.Error())
 		return
 	}
 	connV2 := r.Client.AtlasV2
 	projectID := tfModel.ProjectID.ValueString()
 	clusterName := tfModel.ClusterName.ValueString()
 	if _, _, err := connV2.ClustersApi.GrantMongoDBEmployeeAccess(ctx, projectID, clusterName, atlasReq).Execute(); err != nil {
-		diagnostics.AddError(errorCreate, err.Error())
+		diagnostics.AddError(errorCreateUpdate, err.Error())
+		return
+	}
+	id, err := conversion.IDWithProjectIDClusterName(projectID, clusterName)
```

Comment:
i got errors in the import state tests when ID was not defined:
```

resource_test.go:26: ImportStateVerify: New resource missing identifier attribute "id", ensure attribute value is properly set or use ImportStateVerifyIdentifierAttribute to choose different attribute
```

so that's why added ID

Reply:
https://developer.hashicorp.com/terraform/plugin/framework/acctests#no-id-found-in-attributes

> terraform-plugin-testing version 1.5.0 and later no longer require managed resources and data resources to implement the id attribute.

I remember discussing this with @AgustinBettati not that long ago.

Reply:
I think you should be able to use the `ImportStateVerifyIdentifierAttribute` and set it to something else, e.g., `cluster_name`

Reply:
I believe it is only related to the testing framework. I won't affect ImportState

Reply:
@EspenAlbert cc @AgustinBettati you were right, thanks!  removing ID in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2584/commits/d692acc59a8356760d1726e02c9aee3a04142a4d

Reply:
Yes I agree with Espen, we should be able to use ImportStateVerifyIdentifierAttribute defining an attribute that will uniquely match the imported resource within the context of the test.

Reply:
done, please review again, thx!

Reply:
Nice! Thank you for changing 😁